extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts

```diff
@@ -393,6 +512,78 @@ export class PublishDatabaseDialog {
 		return connectionRow;
 	}
 
+	private createLocalDbInfoRow(view: azdataType.ModelView): azdataType.FlexContainer {
+		this.serverPortTextBox = view.modelBuilder.inputBox().withProps({
+			value: constants.defaultPortNumber,
+			ariaLabel: constants.serverPortNumber,
+			placeHolder: constants.serverPortNumber,
+			width: cssStyles.publishDialogTextboxWidth,
+			enabled: true,
+			inputType: 'number',
+			validationErrorMessage: constants.portMustBeNumber
+		}).withValidation(component => utils.validateSqlServerPortNumber(component.value)).component();
+
+		this.serverPortTextBox.onTextChanged(() => {
+			this.tryEnableGenerateScriptAndOkButtons();
+		});
+		const serverPortRow = this.createFormRow(view, constants.serverPortNumber, this.serverPortTextBox);
```

Comment:
a nice enhancement later on would be to have validation to let the user know if the port is already being used

Reply:
will create a note to do later